test/ecto/query_test.exs

```diff
@@ -97,6 +97,20 @@ defmodule Ecto.QueryTest do
     end
   end
 
+  describe "common table expressions" do
+    test "adds CTEs" do
+      cte1 = from p in "posts1"
+      cte2 = from p in "posts2"
+      query = "posts" |> cte(cte1, as: :cte1, recursive: true) |> cte(cte2, as: :cte2)
+
+      assert query.with.recursive
+      assert query.with.queries |> Enum.at(0) |> Map.fetch!(:as) == :cte1
+      assert query.with.queries |> Enum.at(0) |> Map.fetch!(:query) == cte1
+      assert query.with.queries |> Enum.at(1) |> Map.fetch!(:as) == :cte2
+      assert query.with.queries |> Enum.at(1) |> Map.fetch!(:query) == cte2
```

Comment:
We would need tests for interpolation, to make sure we account for ctes correctly.